packages/flutter/lib/src/material/app.dart

```diff
@@ -108,29 +154,68 @@ class MaterialApp extends StatefulWidget {
   ///
   /// If the app only has one page, then you can specify it using [home] instead.
   ///
-  /// If [home] is specified, then it is an error to provide a route
-  /// in this map for the [Navigator.defaultRouteName] route (`'/'`).
+  /// If [home] is specified, then it implies an entry in this table for the
+  /// [Navigator.defaultRouteName] route (`/`), and it is an error to
+  /// redundantly provide such a route in the [routes] table.
   ///
-  /// If a route is requested that is not specified in this table (or
-  /// by [home]), then the [onGenerateRoute] callback is called to
-  /// build the page instead.
+  /// If a route is requested that is not specified in this table (or by
+  /// [home]), then the [onGenerateRoute] callback is called to build the page
+  /// instead.
   final Map<String, WidgetBuilder> routes;
 
   /// The name of the first route to show.
   ///
-  /// Defaults to [Window.defaultRouteName].
+  /// Defaults to [Window.defaultRouteName], which may be overridden by the code
+  /// that launched the application.
+  ///
+  /// If the route contains slashes, then, before this route is loaded, the
+  /// routes leading to this one are loaded also. For example, if the route was
+  /// `/a/b/c`, then the app would start with the three routes `/a`, `/a/b`, and
+  /// `/a/b/c` loaded, in that order.
+  ///
+  /// If any part of this fails to generate routes, then the [initialRoute] is
+  /// ignored and [Navigator.defaultRouteName] is used instead (`/`). This can
+  /// happen if the app is started with an intent that specifies a non-existent
```

Comment:
Maybe we need to use a generic term, like "deep link", here and link to a TBD explanation of how the idea maps to something concrete on iOS and Android.